client/components/gravatar/index.jsx

```diff
@@ -64,10 +73,18 @@ module.exports = React.createClass( {
 		}
 
 		const alt = this.props.alt || this.props.user.display_name;
-		const avatarURL = this.getResizedImageURL( safeImageURL( this.props.user.avatar_URL ) );
+
+		const avatarURL = this.props.tempImage ||
+			this.getResizedImageURL( safeImageURL( this.props.user.avatar_URL ) );
 
 		return (
 			<img alt={ alt } className="gravatar" src={ avatarURL } width={ size } height={ size } onError={ this.onError } />
 		);
 	}
-} );
+}
+
+export default connect( ( state, ownProps ) => {
+	return {
+		tempImage: getUserTempGravatar( state, get( ownProps, 'user.ID' ) ),
```

Comment:
Do you mean the user passed in to the `<Gravatar />` component? If that's the case, we don't always have an ID or a user object to rely on. I'd rather not complicate the logic, since `getUserTempGravatar` expects a missing ID if we don't have it.